src/workerd/jsg/jsg.h

```diff
@@ -2042,6 +2042,38 @@ class JsMessage;
 
 class DOMException;
 
+// RAII class to adjust the amount of external memory attributed to an isolate.
+// The adjustment will be automatically decremented when the object is destroyed.
+// The allocation amount can be adjusted up or down during the lifetime of an object.
+class ExternalMemoryAdjuster final {
+public:
+  ExternalMemoryAdjuster() = default;
+  ExternalMemoryAdjuster(v8::Isolate* isolate, int64_t amount);
+  ExternalMemoryAdjuster(ExternalMemoryAdjuster&& other);
+  ExternalMemoryAdjuster& operator=(ExternalMemoryAdjuster&& other);
+  KJ_DISALLOW_COPY(ExternalMemoryAdjuster);
+
+  // Increment the amount of external memory reported. We pass in the
+  // Lock& only to prove that we are holding the isolate lock.
+  void increment(Lock&, int64_t amount);
```

Comment:
I think there should be a simple `set()` method too, which calculates the adjustment based on the previous value.

Maybe `increment` and `decrement` can be combined into a single `adjust` and you pass a negative value for decrement?